src/easylink/graph_components.py

```diff
@@ -27,15 +35,83 @@ class Edge:
     target_node: str
     output_slot: str
     input_slot: str
+    filepaths: Optional[tuple[str]] = None
+
+    @classmethod
+    def from_graph_edge(cls, source, sink, edge_attrs) -> "Edge":
+        return cls(
+            source,
+            sink,
+            edge_attrs["output_slot"].name,
+            edge_attrs["input_slot"].name,
+            edge_attrs.get("filepaths"),
+        )
+
+
+class StepGraph(nx.MultiDiGraph):
+    def add_node_from_step(self, step: "Step") -> None:
+        super().add_node(step.name, step=step)
+
+    def add_edge_from_data(self, edge: Edge) -> None:
```

Comment:
If we renamed the `Edge` class, I think this method could be named more descriptively